### PR TITLE
Vanilla Text Classification 

### PR DESCRIPTION
**Pull request by nahidgrid**
## What does this PR do?

Vanilla text classification example without transformers. Using `AG_News` dataset

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_examples/basic_examples/text_classification.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+from argparse import ArgumentParser
+
+import torch
+import torch.nn as nn
+import pytorch_lightning as pl
+from torch.nn import functional as F
+from torch.utils.data import DataLoader, random_split
+from torchvision import transforms
+import torchtext
```
Comment (nateraw) [inline pl_examples/basic_examples/text_classification.py:23]: Do we install torchtext anywhere? Probably needs to be added to examples requirements file.

File: pl_examples/basic_examples/text_classification.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+
+def generate_batch(batch):
+        label = torch.tensor([entry[0] for entry in batch])
+        text = [entry[1] for entry in batch]
+        offsets = [0] + [len(entry) for entry in text]
+        offsets = torch.tensor(offsets[:-1]).cumsum(dim=0)
+        text = torch.cat(text)
+        return text, offsets, label
+
```
Comment (nateraw) [inline pl_examples/basic_examples/text_classification.py:87]:
```suggestion


```

File: pl_examples/basic_examples/text_classification.py (hunk 1 above)
Comment (nahidgrid) [reply pl_examples/basic_examples/text_classification.py:23]: yeah, I installed in my env, will add in the requirements

File: pl_examples/basic_examples/text_classification.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        loss = F.cross_entropy(y_hat, y)
+        self.log('test_loss', loss)
+
+    def configure_optimizers(self):
+        return torch.optim.SGD(self.parameters(), lr=self.hparams.learning_rate)
+
+    @staticmethod
+    def add_model_specific_args(parent_parser):
+        parser = ArgumentParser(parents=[parent_parser], add_help=False)
+        parser.add_argument('--learning_rate', type=float, default=0.0001)
```
Comment (nateraw) [inline pl_examples/basic_examples/text_classification.py:75]: I like that this is just a model specific arg. perhaps add embed dim?

Vocab size + num classes make sense to come from datamodule (so it should stay positional, imo). embed dim could be given a sensible default and then added here as model specific arg.

**Review by nateraw — COMMENTED** [review]
lookin good. I'll have to pull it down and run it before I can comment any further. added some initial thoughts (mostly style stuff).

File: pl_examples/basic_examples/text_classification.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    train_dataset, test_dataset = text_classification.DATASETS['AG_NEWS'](root='./.data', ngrams=NGRAMS, vocab=None)
+    X_train, X_val = random_split(train_dataset, [100000, 20000])
+
+    train_loader = DataLoader(X_train, batch_size=args.batch_size, collate_fn=generate_batch)
+    val_loader = DataLoader(X_val, batch_size=args.batch_size, collate_fn=generate_batch)
+    test_loader = DataLoader(test_dataset, batch_size=args.batch_size, collate_fn=generate_batch)
+
+    VOCAB_SIZE = len(train_dataset.get_vocab())
+    EMBED_DIM = 32
+    NUN_CLASS = len(train_dataset.get_labels())
```
Comment (rohitgr7) [inline pl_examples/basic_examples/text_classification.py:114]: maybe use `LightningDataModule` for this stuff?

Comment (nateraw) [reply pl_examples/basic_examples/text_classification.py:114]: +1 on this. @nahidgrid reach out if ya want help on doin that.
